lightning-net-tokio/src/lib.rs

```diff
@@ -1,277 +1,423 @@
-extern crate bytes;
-extern crate tokio;
-extern crate tokio_codec;
-extern crate futures;
-extern crate lightning;
-extern crate secp256k1;
-
-use bytes::BufMut;
-
-use futures::future;
-use futures::future::Future;
-use futures::{AsyncSink, Stream, Sink};
-use futures::sync::mpsc;
+//! A socket handling library for those running in Tokio environments who wish to use
+//! rust-lightning with native TcpStreams.
+//!
+//! Designed to be as simple as possible, the high-level usage is almost as simple as "hand over a
+//! TcpStream and a reference to a PeerManager and the rest is handled", with the exception of the
+//! act of reading bytes from the remote peer possibly triggering the need to handle some
+//! [Events](../lightning/util/events/enum.Event.html).
```

Comment:
The exception here seems less of an exception and more of the primary way of interacting with the API. I would recommend rewording this to indicate as such.

Reply:
Hmm, that was meant as a "the only thing that *isnt* as trivial is Event handling". Hope its clearer now.